src/Microsoft.AspNetCore.Cors/Infrastructure/CorsService.cs

```diff
@@ -78,12 +79,23 @@ public CorsResult EvaluatePolicy(HttpContext context, CorsPolicy policy)
                 throw new ArgumentNullException(nameof(policy));
             }
 
-            var corsResult = new CorsResult();
-            var accessControlRequestMethod = context.Request.Headers[CorsConstants.AccessControlRequestMethod];
-            if (string.Equals(context.Request.Method, CorsConstants.PreflightHttpMethod, StringComparison.OrdinalIgnoreCase) &&
-                !StringValues.IsNullOrEmpty(accessControlRequestMethod))
+            var origin = context.Request.Headers[CorsConstants.Origin];
+            // A CORS-preflight request ... uses `OPTIONS` as method and includes these headers:
+            // Access-Control-Request-Method, Access-Control-Request-Headers
+            // To keep parity with legacy behavior, we'll ignore the presence of Access-Control-Request-Headers header. 
+            var requestHeaders = context.Request.Headers;
+            var isPreflightRequest =
+                string.Equals(context.Request.Method, CorsConstants.PreflightHttpMethod, StringComparison.OrdinalIgnoreCase) &&
+                requestHeaders.ContainsKey(CorsConstants.AccessControlRequestMethod);
```

Comment:
Why did we switch from checking IsNullOrEmpty to just ContainsKey? I imagine that `Access-Control-Request-Method:` will fail later down the pipe, but just checking.

I do believe that Access-Control-Request-Headers is not mandatory, it's probably why it's not being check there.

I think the overall comment above might be misleading (or at least I don't understand it). It seems to imply that we are changing the behavior, but I don't see how the behavior is different.

Reply:
I haven't seen a browser request without the header, but it really isn't a hill to die on 👍

Reply:
> Why did we switch from checking IsNullOrEmpty to just ContainsKey?

It seems enough for us to verify the presence of the header - we don't actually use the header value here. If you feel strongly about it, I can add it back. Removed the comment, I agree it didn't help.